Integrate Docker and enhance mobile build configuration

mael-app opened the pull request:
## Description
This PR introduces Docker integration for production deployment and improves the mobile application's build configuration. Key changes include:

- **Docker Integration**:
  - Added Dockerfiles for backend, frontend, and mobile apps.
  - Configured `docker-compose.yml` for production services (PostgreSQL, Redis).
  - Introduced healthchecks, dynamic environment variable setup, and centralized NGINX configuration.
  - Added a validation script, `validate-docker.sh`, to test service health.
  - Documentation for deployment setup included in `DOCKER_DEPLOYMENT.md`.

- **Mobile Build Improvements**:
  - Migrated hardcoded configurations to `.env` files for backend URLs and OAuth settings.
  - Added commands to simplify APK and AAB builds.
  - Updated dependencies and added `babel.config.js` with required Expo presets and plugins.
  - Updated `.gitignore` to exclude `.env` files.

These changes aim to streamline production deployment and enhance mobile app development workflows.

## Type of Change
- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please specify)

## Testing
List the tests performed to verify your changes. Include instructions to reproduce if needed.

- [ ] Unit tests
- [x] Manual tests
- [x] Integration tests (if applicable)
---

mael-app committed:
feat<mobile>: migrate configuration to `.env` and enhance build setup

- Replaced hardcoded values with `.env` file for backend URL and OAuth configuration.
- Added commands to simplify APK and AAB builds for Android.
- Updated dependencies and configuration files to support the new build and runtime setup.
- Introduced `babel.config.js` with Expo preset and necessary plugins.
- Updated `.gitignore` to exclude `.env` files.

mael-app committed:
feat<docker>: integrate Docker for production deployment

- Added Dockerfiles for backend, frontend, and mobile apps.
- Configured `docker-compose.yml` for production services, including PostgreSQL and Redis.
- Introduced healthchecks, dynamic environment variable setup, and entrypoints.
- Implemented APK build process for mobile and centralized NGINX configuration for the web app.
- Added instructions in `DOCKER_DEPLOYMENT.md` for deployment setup.
- Included validation script `validate-docker.sh` for testing service health.

mael-app committed:
feat<docker>: add environment variables for API and OAuth configuration

- Included ARG and ENV instructions in mobile and frontend Dockerfiles for injecting API URL and OAuth-related values dynamically.

Watersrc-stack commented:
`validate-docker.sh` script introduced on commit `ce1ebde` seems to no function at all.
<img width="1253" height="372" alt="image" src="https://github.com/user-attachments/assets/096f0544-9579-48a7-bb4e-f58eb7e7e117" />

mael-app committed:
feat<frontend>: implement centralized API fetch utility and enhance proxy configuration

- Added `fetch.ts` utility for handling API calls with automatic base URL and error handling.
- Updated `tsconfig.json` to include `vite/client` types and `vite-env.d.ts` for environment typing.
- Refactored `googleAuth.ts` to use the new `apiFetch` utility.
- Simplified `baseURL` logic in `api.ts`, ensuring compatibility across environments.
- Enhanced `nginx.conf` to handle specific API routes and improve frontend/backend integration.
- Modified Dockerfile to streamline APK handling process during app initialization.

mael-app committed:
feat<mobile>: enhance config validation and error handling

- Added fallback for `API_URL` with a warning if not set in `.env` file.
- Introduced runtime validation for critical configuration variables.
- Enhanced debugging with detailed logs in development environment.
- Improved network error logging and response interceptor functionality in `api.js`.
- Unified error handling for unauthorized requests and network issues.

mael-app committed:
feat<mobile>: enhance Dockerfile with improved Java, NDK, and build setup

- Dynamically set `JAVA_HOME` for improved compatibility.
- Added NDK and CMake installations to support native builds.
- Enhanced PATH setup for better toolchain accessibility.
- Introduced Gradle optimizations for improved build performance.
- Verified NDK and CMake presence during image build for reliability.

mael-app committed:
feat<mobile>: simplify Dockerfile with prebuilt React Native Android image

- Replaced custom base image with `reactnativecommunity/react-native-android` for streamlined setup.
- Refactored file copy operations for optimized build process.
- Enhanced Gradle performance settings with additional parallelism and caching options.
- Removed redundant NDK and SDK installations, leveraging the prebuilt environment.

mael-app committed:
feat<mobile>: update Dockerfile to include styles.js and config.js

- Added `styles.js` and `config.js` to the file copy operations for completeness.
- Ensures all necessary files are included in the Docker build process.
